src/Elastic.Apm.Azure.Functions/ApmMiddleware.cs

```diff
@@ -114,6 +136,9 @@ private static void SetTriggerSpecificResult(ITransaction transaction, bool succ
 		}
 	}
 
+	private static Dictionary<string, string> CreateHeadersDictionary(HttpHeadersCollection httpHeadersCollection) =>
+		httpHeadersCollection.ToDictionary(h => h.Key, h => string.Join(',', h.Value));
```

Comment:
This needs to be sanitized. We have the [`SanitizeFieldNames`](https://www.elastic.co/guide/en/apm/agent/dotnet/current/config-core.html#config-sanitize-field-names) config, I think that should apply here as well.

[Here is](https://github.com/elastic/apm-agent-dotnet/blob/main/src/Elastic.Apm.AspNetCore/WebRequestTransactionCreator.cs#L126-L132) how we do this in case of ASP.NET Core.

Reply:
I actually thought about sanitization, but then I saw [this code](https://github.com/elastic/apm-agent-dotnet/blob/5672eb4b9e1cd3508e0c3827359374f9cf5459fd/src/Elastic.Apm/Report/PayloadSenderV2.cs#L131) and thought sanitization would happen generically?
(I forgot to test this however tbh.)

Reply:
Hahh, you are right! 

...which means then we should remove it in `WebRequestTransactionCreator.cs` (unrelated to this PR, so no need to follow up here).

Sorry for the confusion.

Reply:
I also saw that and suspected it was a remnant. I'll put it on my TODO list as a follow-up task 👍 .